oxenstored/connection.ml

```diff
@@ -197,6 +236,8 @@ let watch_create ~con ~path ~token =
   ; pending_watchevents=
```

Comment:
Couldn't the type be made opaque and encapsulated in a modeule so all the logic handling this is localized?

Reply:
Sure, I could do that, but there's code that separately calls `Xb.queue` and `Xb.output` anyway so I couldn't encapsulate it all.

Reply:
There is also a bit of gymnastics required due to the order of definition of watch->connection->update_* so that the interface will not be nice